Automattic/jetpack
# Social Post UI: Add trigger and modal to the editor sidebar

## Proposed changes:

* Add the trigger and the modal needed for Social post UI

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the editor preview feature - `define( 'JETPACK_SOCIAL_HAS_EDITOR_PREVIEW', true );`
* Goto post editor
* Open Jetpack or Social sidebar
* Under "Share this post" section, confirm that you see the "Create custom posts" button
* Click on it
* Confirm that the "Social previews" modal opens

Trigger
<img width="289" alt="Screenshot 2024-08-01 at 11 57 07 AM" src="https://github.com/user-attachments/assets/f720ce48-52e4-4730-838b-dfec2abdcac0">

Modal

<img width="1341" alt="image" src="https://github.com/user-attachments/assets/93ad48cd-4f75-4ca4-8d87-e40fa239264a">



https://github.com/user-attachments/assets/b05023c4-43b2-49c9-8a80-e46bf87daaa2
